subprojects/docs/src/samples/userguide/tasks/incrementalBuild/customTaskClass/groovy/build.gradle

```diff
@@ -36,50 +36,58 @@ task processTemplatesAdHoc {
 }
 // end::ad-hoc-task[]
 
+// tag::ad-hoc-task-skip-when-empty[]
+task processTemplatesAdHocSkipWhenEmpty {
+    // ...
+
+    inputs.files(fileTree("src/templates") {
+            include "**/*.fm"
+        })
+        .skipWhenEmpty()
+        .withPropertyName("sourceFiles")
+        .withPathSensitivity(PathSensitivity.RELATIVE)
+
+    // ...
+// end::ad-hoc-task-skip-when-empty[]
+    inputs.property("engine", TemplateEngineType.FREEMARKER)
+    inputs.property("templateData.name", "docs")
+    inputs.property("templateData.variables", [year: 2013])
+    outputs.dir("$buildDir/genOutput2")
+        .withPropertyName("outputDir")
+
+    doLast {
+        copy {
```

Comment:
The `// Process the templates here` comment is missing, while present in the `.kts` file.

Reply:
This part of the code is not shown anywhere to users, I removed the comment from the `.kts` file.

Reply:
Removed in e4894b6ee5a32f78d36033116206f55effdee9e9.